collector/cpufreq_linux.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nocpufreq
+
+package collector
+
+import (
+	"path/filepath"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	cpufreqSubsystem = "cpufreq"
+)
+
+type cpufreqCollector struct {
+	curFreq         *prometheus.Desc
+	minFreq         *prometheus.Desc
+	maxFreq         *prometheus.Desc
+	coreThrottle    *prometheus.Desc
+	packageThrottle *prometheus.Desc
+}
+
+func init() {
+	Factories["cpufreq"] = NewCpufreqCollector
+}
+
+// NewEdacCollector returns a new Collector exposing edac stats.
+func NewCpufreqCollector() (Collector, error) {
+	return &cpufreqCollector{
+		curFreq: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, cpufreqSubsystem, "current_frequency_hertz"),
```

Comment:
We don't usually put "current" into gauge names, that's implied. This would be the first time in node_exporter we'd be doing that, so for consistency I'd remove it, but I see that it follows the naming pattern around minimum and maximum. Suggestions @brian-brazil?